### PR TITLE
[Merged by Bors] - chore: generalize theorems from IsConnected to IsPreconnected

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/IsConnected.lean [hunk 1]
```diff
@@ -318,35 +337,35 @@ theorem zag_of_zag_obj (F : J ⥤ K) [Full F] {j₁ j₂ : J} (h : Zag (F.obj j
...
+    induct_on_objects {k | r j₁ k} (hr.1 j₁)
       fun f => ⟨fun t => hr.3 t (h f), fun t => hr.3 t (hr.2 (h f))⟩
-  intros
-  apply hr.3 (hr.2 (z _)) (z _)
+  exact z j₂
 #align category_theory.equiv_relation CategoryTheory.equiv_relation
 
 /-- In a connected category, any two objects are related by `Zigzag`. -/
-theorem isConnected_zigzag [IsConnected J] (j₁ j₂ : J) : Zigzag j₁ j₂ :=
+theorem isConnected_zigzag [IsPreconnected J] (j₁ j₂ : J) : Zigzag j₁ j₂ :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/IsConnected.lean:350]:
```suggestion
theorem isPreconnected_zigzag [IsPreconnected J] (j₁ j₂ : J) : Zigzag j₁ j₂ :=
```

File: Mathlib/CategoryTheory/IsConnected.lean [hunk 2]
```diff
@@ -398,7 +427,7 @@ instance [IsConnected J] : Full (Functor.const J : C ⥤ J ⥤ C) where
     ext j
     apply nat_trans_from_is_connected f (Classical.arbitrary J) j
 
-instance nonempty_hom_of_connected_groupoid {G} [Groupoid G] [IsConnected G] :
+instance nonempty_hom_of_connected_groupoid {G} [Groupoid G] [IsPreconnected G] :
```
Comment (joelriou) [inline Mathlib/CategoryTheory/IsConnected.lean:430]:
```suggestion
instance nonempty_hom_of_preconnected_groupoid {G} [Groupoid G] [IsPreconnected G] :
```

Comment (TwoFX) [reply Mathlib/CategoryTheory/IsConnected.lean]: Thanks, fixed.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**